lib/config.js

```diff
@@ -5,24 +5,37 @@ const isPlainObject = require('is-plain-obj');
 const TimeFixPlugin = require('time-fix-plugin');
 const { MultiCompiler } = require('webpack');
 
+function entryFixValue(entry) {
+  let result = entry;
+
+  if (typeof result === 'string') {
+    result = [result];
+  } else if (isPlainObject(result)) {
+    for (const key of Object.keys(result)) {
+      const value = result[key];
+
+      if (!Array.isArray(value)) {
+        result[key] = [value];
+      }
+    }
+  }
+
+  return result;
+}
+
 // automagically wrap hot-client-invalid entry values in an array
 function toArray(config) {
   const result = config;
 
-  if (typeof result.entry === 'string') {
-    result.entry = [result.entry];
-  } else if (typeof result.entry === 'undefined') {
+  if (typeof result.entry === 'undefined') {
     // webpack v4 defaults an empty config to { entry: './src' }. but since we
     // need an array, we'll mimic that default config.
     result.entry = ['./src'];
-  } else if (isPlainObject(result.entry)) {
-    for (const key of Object.keys(result.entry)) {
-      const entry = result.entry[key];
-
-      if (!Array.isArray(entry)) {
-        result.entry[key] = [entry];
-      }
-    }
+  } else if (result.entry instanceof Function) {
```

Comment:
There's no need for a separate function to be broken out of `toArray`. Please restore the original function signature, add your function handling block, and simply make a call to itself.

Reply:
Not clear what you mean by "function signature". The behavior of toArray should not have changed (note the else on line 38).

In any case, I've dropped these tools, so won't waste any more of my time.